File: weektwo/questions/Bubble.java
package questions;


//Given an array of integers nums, find the kth smallest element in the array using Bubble Sort.
//nums = [4, 2, 1, 3], k = 2
// Output: 2

public class Bubble {
    public static void main(String[] args) {
        // Example usage:
        int[] nums = { 4, 2, 1, 3 };
        int k = 2;
        int kthSmallest = findKthSmallest(nums, k);
        System.out.println(kthSmallest); // Output: 2
    }

    public static int findKthSmallest(int[] nums, int k) {
        int n = nums.length;
        for (int i = 0; i < n - 1; i++) {
            for (int j = 0; j < n - i - 1; j++) {
                if (nums[j] > nums[j + 1]) {
                    int temp = nums[j];
                    nums[j] = nums[j + 1];
                    nums[j + 1] = temp;
                }
            }
            if (i == k - 1) {
                return nums[i];
            }
        }
        return nums[k - 1];
    }

}

File: dsaJavascript/toi/bubble.js
function bubble(nums) {
  for (let i = 0; i < nums.length - 1 ; i++) {
    let swapped = false;
    for (let j = 0; j < nums.length - i - 1; j++) {
      if (nums[j] > nums[j + 1]) {
        swap(nums, j, j + 1);
        swapped = true;
      }
    }
    if (!swapped) break;
  }
}

function swap(nums, i, j) {
  let temp = nums[i];
  nums[i] = nums[j];
  nums[j] = temp;
}

let nums = [119, 22, 12, 9, 11, 2];

bubble(nums);

console.log(nums);
